neps/nep-0488.md

```diff
@@ -0,0 +1,1223 @@
+---
+NEP: 488
+Title: Host Functions for BLS12-381 Curve Operations
+Authors: Olga Kuniavskaia <olga.kunyavskaya@aurora.dev>
+Status: Draft
+DiscussionsTo: https://github.com/nearprotocol/neps/pull/488
+Type: Runtime Spec
+Version: 0.0.1
+Created: 2023-07-17
+LastUpdated: 2023-11-21
+---
+
+## Summary
+
+This NEP introduces host functions to perform operations on the BLS12-381 elliptic curve. It is a minimal set of functions needed to efficiently verify BLS signatures and zkSNARKs.
+
+## Motivation
+
+The primary aim of this NEP is to enable fast and efficient verification of BLS signatures and zkSNARKs based on the BLS12-381[^1],[^11],[^52] elliptic curve on NEAR.
+
+To efficiently verify zkSNARKs[^19], host functions for operations on the BN254 
+elliptic curve (also known as Alt-BN128)[^9], [^12] have already been implemented on NEAR[^10]. 
+For instance, the Zeropool[^20] project utilizes these host functions for verifying zkSNARKs on NEAR. 
+However, recent research shows that the BN254 security level is lower than 100-bit[^13] and it is not recommended for use. 
+BLS12-381, on the other hand, offers over 120 bits of security[^8] and is widely used[^2],[^3],[^4],[^5],[^6],[^7]  as a robust alternative. 
```

Comment:
I think that commas shouldn't be used between multiple consecutive references (for example, Wikipedia doesn't use them).

Reply:
Sure. But I couldn't beat the automatic style checks in git. If you do not put commas, then it will think that everything except the first link is not used, and will be unhappy with unused references. That's why I preferred to put commas so that there would be a green check mark in git.